common/scripted_triggers/wc_religious_triggers.txt

```diff
@@ -1155,4 +1155,19 @@ has_saint_relics_trigger = {
 		religion = religion:light_group
 		this = faith:tidemother
 	}
+}
+
+no_mercenaries_trigger = {
+	# Fundamentalist dark faiths do not tolerate heretics
+	scope:character = {
+		religion = {
```

Comment:
There shouldn't be set scopes in scripted triggers. Otherwise, it won't work in environment where these scopes aren't set.

Also, you scope to character then to their religion. The only place where you used this trigger scopes to character then to their religion too. So the full scoping path looks: character > religion > character > religion.

That's how you should rewrite it:
```
no_mercenaries_faith_trigger = {
	evil_public_religion_trigger = yes
}
```
Faith in trigger name indicates it should be used in faith scope. Also, using `evil_public_religion_trigger` will be easier as it includes only the Scourge, Legion, Black Empire.
The using should look like this:
```
scope:character = { faith = { no_mercenaries_faith_trigger  = yes } }
```